pkgs/top-level/splice.nix

```diff
@@ -151,6 +151,8 @@ in
 
   newScope = extra: lib.callPackageWith (pkgsForCall // extra);
 
+  pkgs = if actuallySplice then splicedPackages // { recurseForDerivations = false; } else pkgs;
```

Comment:
@Ericson2314 
I wonder why you didn't do this instead of adding `__splicedPackages`? https://github.com/NixOS/nixpkgs/commit/f27f491784cd19abca6494ae28306deeea17c613